src/Api/Helper.php

```diff
@@ -28,6 +31,85 @@ public function __construct(Api $api)
     }
 
     /**
+     * Upload file
+     *
+     * Returns file key to use for list import, email content, etc.
+     *
+     * @param string $path
+     * @return string file key
+     */
+    public function uploadFile($path)
+    {
+        if (!is_readable($path)) {
+            throw new \InvalidArgumentException('File path is not readable: ' . $path);
+        }
+        $basename = basename($path);
+        $mime = (new \finfo(FILEINFO_MIME_TYPE))->file($path);
+        if ($mime === false) {
+            throw new \RuntimeException("MIME type could not be determined");
+        }
+
+        // Initialise
+        /** @var string $fileKey */
+        $fileKey = $this->api->file_upload->initialise()->key;
+
+        // Build request
+        $this->setConnectionConfig($this->api->getConfig());
+        $headers = $this->getHeaders();
+
+        $handleurl = ($this->useSsl ? 'https' : 'http') .
+            '://' .
+            $this->host .
+            '/api/json/file_upload';
+
+        $params = array(
+            'method' => 'handle',
+            'key'    => $fileKey,
+            'file'   => "@{$path};filename={$basename};type={$mime}"
```

Comment:
According to [this comment](http://php.net/manual/en/class.curlfile.php#115161) and [this page](http://php.net/manual/en/function.curl-setopt.php) in the PHP docs (search for 'CURLOPT_POSTFIELDS'), there are issues with  using the '@' approach to file uploads, and it is deprecated from PHP5.5.

I can't actually find anywhere saying what those issues are though..

Apparently the solution is to use [CurlFile](http://php.net/manual/en/class.curlfile.php) instead.